src/account/tests.py

```diff
@@ -1,3 +1,243 @@
+
+        self.client.logout()
+
+    # -------------------------------------------------------------------------
+    # DELETE
+    # -------------------------------------------------------------------------
+
+    def test_delete_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        del_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(del_idx), '/'])
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+        self.client.logout()
+
+    # -------------------------------------------------------------------------
+    # SINGLE view
+    # -------------------------------------------------------------------------
+
+    # -------------------------------------------------------------------------
+    # GET
+    # -------------------------------------------------------------------------
+
+    def test_get_single_random_admin(self):
+        self.client.force_login(User.objects.get(username='admin'))
+        get_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_random_mod(self):
+        self.client.force_login(User.objects.get(username='mod'))
+        get_idx = Profile.objects.order_by('?').first().pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        profile = Profile.objects.get(pk=get_idx)
+        serializer = ProfileSerializer(profile)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+        self.client.logout()
+
+    def test_get_single_not_own_user(self):
+        self.client.force_login(User.objects.get(username='user'))
+        get_idx = User.objects.get(username='other_user').pk
+        url = ''.join([reverse('profile-list'), str(get_idx), '/'])
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
I think users should be able to view other users' profiles. This is a function on the original StackOverflow, or imagine visiting someone's profile on Facebook.